grant-project/grant-monorepo
# EIP-712 signatures for login & signup

Closes #181, closes #183.

### What This Does

* Implements the full EIP-712 spec for message signing.
* Reworks signatures in redux / api to include raw data.
* Makes use of @dternyak's EIP-712 validating service to check them on the backend for POST /user signup and newly added POST /user/auth login.

### Steps to Test

1. Create a new account, confirm signature was required and it worked.
2. Sign out, then sign back in, confirm signature was required and it worked.
3. Refresh page, confirm you were logged back in.
4. Try canceling signature prompt for steps 1 & 2, confirm you got an error.
5. Load the page as one metamask account, and then switch to another.

I'm unsure as to how you could easily test an "invalid" signature, but I've manually edited the code to try it out and it seems to work.

## Screenshots

<img width="472" alt="screen shot 2018-11-06 at 4 05 50 pm" src="https://user-images.githubusercontent.com/649992/48094891-5123ec00-e1e1-11e8-8d8e-41a25eb29c92.png">
